CMake/Macros.cmake

```diff
@@ -29,8 +29,19 @@
 # order of least-to-most-dependent.  Some libraries depend on others
 # to link correctly.
 macro(check_library_exists_concat LIBRARY SYMBOL VARIABLE)
-  check_library_exists("${LIBRARY};${CURL_LIBS}" ${SYMBOL} "${CMAKE_LIBRARY_PATH}"
+  # Enumerate CURL_LIBS and add each library to CMAKE_REQUIRED_LIBRARIES if it
+  # is not a CMake target. System libraries should not depend on CMake targets.
+  set(CMAKE_REQUIRED_LIBRARIES)
+  foreach(LIB ${CURL_LIBS})
+    if (NOT TARGET ${LIB})
+      set(CMAKE_REQUIRED_LIBRARIES ${LIB} ${CMAKE_REQUIRED_LIBRARIES})
+    endif()
+  endforeach()
```

Comment:
I think the problem is actually that `CURL_LIBS` is not maintained only by `check_library_exists_concat` anymore.  Other code paths are adding to that list too.  Instead of filtering the list here on every call, I think it will be better to refactor things so a separate list is used for accumulation by `check_library_exists_concat` calls.  One possibility is to keep `CURL_LIBS` here and update all other sites that add to it to use another list or directly call `target_link_libraries` for the dependents.

Reply:
Yes, but curl's code is what adds `ZLIB::ZLIB` to `CURL_LIBS`.  It's not the result of a `check_library_exists_concat` call, and so should not be added to the list that `check_library_exists_concat` is accumulating.  It should be linked by other means.

Reply:
Side note: `OpenSSL::SSL OpenSSL::Crypto` aliases are also used by curl.

Reply:
ldap+lber detection may depend on OpenSSL, and at that point we already detected OpenSSL. Thus it seems feasible to use `set(CMAKE_REQUIRED_LIBRARIES "${OPENSSL_LIBRARIES}")` when checking for ldap+lber.

Rest of checks don't need any dependency detected by curl.

Reply:
In that case, if OpenSSL is an CMake `ALIAS`, how has that worked in the past? Seems like it would fail just like in our `ZLIB::ZLIB` case. I imagine nobody has ever run into that scenario because most people have some variety of OpenSSL already installed on their system (so why would they want to compile OpenSSL statically).

Reply:
Yes, I'm guessing in most cases OpenSSL is already built. I never tried building curl from an upper-level CMake (or make it build dependencies in a single run), but in such cases, projects need to be built in reverse topological order of dependency to work correctly.

Reply:
@bradking I have opened a new PR for the changes you suggested. #11648